### PR TITLE
Target_Blank option to open External links

### PR DESCRIPTION
**Pull request by nickvergessen**
Rebased and fixed version of https://github.com/nextcloud/server/pull/27725

### CODE REVIEW

File: core/templates/layout.user.php [hunk 1]
```diff
@@ -69,6 +69,7 @@
 					<?php foreach ($_['navigation'] as $entry): ?>
 						<li data-id="<?php p($entry['id']); ?>" class="hidden" tabindex="-1">
 							<a href="<?php print_unescaped($entry['href']); ?>"
+								<?php if ($entry['target']): ?> target="_blank" rel="noreferrer noopener"<?php endif; ?>
```
Comment (artonge) [inline core/templates/layout.user.php:72]: Not sure where $entry['target'] comes from, but can it be something else than `blank` ?

```suggestion
								<?php if ($entry['target'] === "blank"): ?> target="_blank" rel="noreferrer noopener"<?php endif; ?>
```

**Review by artonge — APPROVED** [review]

File: core/templates/layout.user.php (hunk 1 above)
Comment (nickvergessen) [reply core/templates/layout.user.php:72]: It's new, so no.
Boolean sounds more sane to reduce issues?

https://github.com/nextcloud/external/pull/287

Comment (artonge) [reply core/templates/layout.user.php:72]: Ah, thanks. If it's new then yes, let's go for boolean, but maybe with a more explicit name?

Comment (artonge) [reply core/templates/layout.user.php:72]: `target_blank` would be enough

**Review by vitormattos — APPROVED** [review]
I tested locally, working fine.

**Comment by CarlSchwan** [comment]
I'm getting some strange error on master now:

```
nextcloud-nextcloud-1  | {"reqId":"rm6ZRsYKVIFSetdRMIsU","level":3,"time":"2022-02-16T11:18:15+00:00","remoteAddr":"192.168.21.5","user":"admin","app":"PHP","method":"GET","url":"/index.php/settings/apps","message":"Undefined index: target at /var/www/html/core/templates/layout.user.php#109","userAgent":"Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:93.0) Gecko/20100101 Firefox/93.0","version":"24.0.0.4","exception":{"Exception":"Error","Message":"Undefined index: target at /var/www/html/core/templates/layout.user.php#109","Code":0,"Trace":[{"file":"/var/www/html/core/templates/layout.user.php","line":109,"function":"onAll","class":"OC\\Log\\ErrorHandler","type":"::","args":[8,"Undefined index: target","/var/www/html/core/templates/layout.user.php",109,{"file":"/var/www/html/core/templates/layout.user.php","additionalParams":{"serverData":{"appstoreEnabled":true,"updateCount":0,"developerDocumentation":"https://docs.nextcloud.com/server/24/go.php?to=developer-manual","bundles":[{"name":"Enterprise bundle","id":"EnterpriseBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Hub bundle","id":"HubBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Groupware bundle","id":"GroupwareBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Social sharing bundle","id":"SocialSharingBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Education Edition","id":"EducationBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]}]}},"_":{"serverData":{"appstoreEnabled":true,"updateCount":0,"developerDocumentation":"https://docs.nextcloud.com/server/24/go.php?to=developer-manual","bundles":[{"name":"Enterprise bundle","id":"EnterpriseBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Hub bundle","id":"HubBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Groupware bundle","id":"GroupwareBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Social sharing bundle","id":"SocialSharingBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]},{"name":"Education Edition","id":"EducationBundle","appIdentifiers":["Encoding skipped as the maximum nesting level was reached"]}]},"requesttoken":"U8nKeQjmFMso1+2X5hm/ZiqLYbpzB4ZazXkB5j+HwVY=:HfuDVk+2QL1rnY/F0HbYDHu4J/MqbNJtphhStHPT8DQ=","bodyid":"body-settings","logoUrl":"","application":"Apps","0":"And 19 more entries, set log level to debug to see all entries"},"l":{"__class__":"OC\\L10N\\LazyL10N"},"theme":{"__class__":"OCP\\Defaults"},"0":"And 30 more entries, set log level to debug to see all entries"}]},{"file":"/var/www/html/lib/private/Template/Base.php","line":180,"args":["/var/www/html/core/templates/layout.user.php"],"function":"include"},{"file":"/var/www/html/lib/private/Template/Base.php","line":150,"function":"load","class":"OC\\Template\\Base","type":"->","args":["/var/www/html/core/templates/layout.user.php",{"serverData":{"appstoreEnabled":true,"updateCount":0,"developerDocumentation":"https://docs.nextcloud.com/server/24/go.php?to=developer-manual","bundles":[{"name":"Enterprise bundle","id":"EnterpriseBundle","appIdentifiers":["admin_audit","user_ldap","files_retention","files_automatedtagging","user_saml","And 2 more entries, set log level to debug to see all entries"]},{"name":"Hub bundle","id":"HubBundle","appIdentifiers":["spreed","contacts","calendar","mail","richdocuments","And 1 more entries, set log level to debug to see all entries"]},{"name":"Groupware bundle","id":"GroupwareBundle","appIdentifiers":["calendar","contacts","deck","mail"]},{"name":"Social sharing bundle","id":"SocialSharingBundle","appIdentifiers":["socialsharing_twitter","socialsharing_facebook","socialsharing_email","socialsharing_diaspora"]},{"name":"Education Edition","id":"EducationBundle","appIdentifiers":["dashboard","circles","groupfolders","announcementcenter","quota_warning","And 1 more entries, set log level to debug to see all entries"]}]}}]},{"file":"/var/www/html/lib/private/legacy/OC_Template.php","line":182,"function":"fetchPage","class":"OC\\Template\\Base","type":"->","args":[{"serverData":{"appstoreEnabled":true,"updateCount":0,"developerDocumentation":"https://docs.nextcloud.com/server/24/go.php?to=developer-manual","bundles":[{"name":"Enterprise bundle","id":"EnterpriseBundle","appIdentifiers":["admin_audit","user_ldap","files_retention","files_automatedtagging","user_saml","And 2 more entries, set log level to debug to see all entries"]},{"name":"Hub bundle","id":"HubBundle","appIdentifiers":["spreed","contacts","calendar","mail","richdocuments","And 1 more entries, set log level to debug to see all entries"]},{"name":"Groupware bundle","id":"GroupwareBundle","appIdentifiers":["calendar","contacts","deck","mail"]},{"name":"Social sharing bundle","id":"SocialSharingBundle","appIdentifiers":["socialsharing_twitter","socialsharing_facebook","socialsharing_email","socialsharing_diaspora"]},{"name":"Education Edition","id":"EducationBundle","appIdentifiers":["dashboard","circles","groupfolders","announcementcenter","quota_warning","And 1 more entries, set log level to debug to see all entries"]}]}}]},{"file":"/var/www/html/lib/private/legacy/OC_Template.php","line":213,"function":"fetchPage","class":"OC_Template","type":"->","args":[{"serverData":{"appstoreEnabled":true,"updateCount":0,"developerDocumentation":"https://docs.nextcloud.com/server/24/go.php?to=developer-manual","bundles":[{"name":"Enterprise bundle","id":"EnterpriseBundle","appIdentifiers":["admin_audit","user_ldap","files_retention","files_automatedtagging","user_saml","And 2 more entries, set log level to debug to see all entries"]},{"name":"Hub bundle","id":"HubBundle","appIdentifiers":["spreed","contacts","calendar","mail","richdocuments","And 1 more entries, set log level to debug to see all entries"]},{"name":"Groupware bundle","id":"GroupwareBundle","appIdentifiers":["calendar","contacts","deck","mail"]},{"name":"Social sharing bundle","id":"SocialSharingBundle","appIdentifiers":["socialsharing_twitter","socialsharing_facebook","socialsharing_email","socialsharing_diaspora"]},{"name":"Education Edition","id":"EducationBundle","appIdentifiers":["dashboard","circles","groupfolders","announcementcenter","quota_warning","And 1 more entries, set log level to debug to see all entries"]}]}}]},{"file":"/var/www/html/lib/public/AppFramework/Http/TemplateResponse.php","line":204,"function":"fetchPage","class":"OC_Template","type":"->","args":[{"serverData":{"appstoreEnabled":true,"updateCount":0,"developerDocumentation":"https://docs.nextcloud.com/server/24/go.php?to=developer-manual","bundles":[{"name":"Enterprise bundle","id":"EnterpriseBundle","appIdentifiers":["admin_audit","user_ldap","files_retention","files_automatedtagging","user_saml","And 2 more entries, set log level to debug to see all entries"]},{"name":"Hub bundle","id":"HubBundle","appIdentifiers":["spreed","contacts","calendar","mail","richdocuments","And 1 more entries, set log level to debug to see all entries"]},{"name":"Groupware bundle","id":"GroupwareBundle","appIdentifiers":["calendar","contacts","deck","mail"]},{"name":"Social sharing bundle","id":"SocialSharingBundle","appIdentifiers":["socialsharing_twitter","socialsharing_facebook","socialsharing_email","socialsharing_diaspora"]},{"name":"Education Edition","id":"EducationBundle","appIdentifiers":["dashboard","circles","groupfolders","announcementcenter","quota_warning","And 1 more entries, set log level to debug to see all entries"]}]}}]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":171,"function":"render","class":"OCP\\AppFramework\\Http\\TemplateResponse","type":"->","args":[]},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Settings\\Controller\\AppSettingsController"},"viewApps"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Settings\\Controller\\AppSettingsController","viewApps",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"category":"","id":"","_route":"settings.AppSettings.viewApps"}]},{"file":"/var/www/html/lib/base.php","line":1009,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/settings/apps"]},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/lib/private/Log/ErrorHandler.php","Line":99,"CustomMessage":"--"}}
```

**Pull request merged**